src/VisualStudio/Core/Impl/CodeModel/AbstractProjectCodeModel.cs

```diff
@@ -84,7 +84,28 @@ public bool TryGetCachedFileCodeModel(string fileName, out ComHandle<EnvDTE80.Fi
             return GetCodeModelCache().GetOrCreateFileCodeModel(filePath);
         }
 
+        public ComHandle<EnvDTE80.FileCodeModel2, FileCodeModel> GetOrCreateFileCodeModel(string filePath, object parent)
+        {
+            return GetCodeModelCache().GetOrCreateFileCodeModel(filePath, parent);
```

Comment:
nit, consider =>

Reply:
can GetCodeModelCache return null?  from other code, it's not clear.  but it makes me worry about thsi code.

Reply:
So GetCodeModelCache() _could_ return null, but it was generally not handled. And the only scenario it could happen in would be if you somehow have a code model infrastructure setup for a project that isn't in the workspace, and couldn't even be pushed into the workspace when we tried to bypass batching. My strong suspicion here is this got refactored many times and as a result landed into this but the result doesn't actually make sense. I applied "when in rome" here and then hope to delete some of this in a future PR.